src/pages/Order/Page.tsx

```diff
@@ -0,0 +1,136 @@
+  useEffect(() => {
+    if (!product) {
+      navigate(ROUTE_PATH.HOME);
+    }
+  }, [product, navigate]);
+
+  const [selectedCard, setSelectedCard] = useState(orderCardMock[0]);
+
+  const messageInput = useStringInput("", getMessageError);
+  const senderInput = useStringInput("", getNameError);
+  const recipientNameInput = useStringInput("", getNameError);
+  const recipientPhoneInput = useStringInput("", getPhoneError);
+  const recipientQuantityInput = useNumberInput(1, getQuantityError);
+
+  const totalPrice = product ? product.price.sellingPrice * recipientQuantityInput.value : 0;
+
+  const handleOrderSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const isValidMessage = messageInput.validate();
+    const isValidSender = senderInput.validate();
+    const isValidRecipientName = recipientNameInput.validate();
+    const isValidRecipientPhone = recipientPhoneInput.validate();
+    const isValidRecipientCount = recipientQuantityInput.validate();
+
+    const isValidOrder =
+      isValidMessage && isValidSender && isValidRecipientName && isValidRecipientPhone && isValidRecipientCount;
+
+    if (isValidOrder) {
+      alertOrderInfo(messageInput.value, product?.name as string, recipientQuantityInput.value, senderInput.value);
+      navigate(ROUTE_PATH.HOME);
+    }
+    console.log(recipientQuantityInput.value, recipientQuantityInput.errorMsg, recipientQuantityInput.validate());
+  };
+
+  useEffect(() => {
+    messageInput.setValue(selectedCard.defaultTextMessage);
+  }, [selectedCard]);
+  return (
+    <Container>
+      <Content onSubmit={handleOrderSubmit}>
+        <Card
+          selectedCard={selectedCard}
+          setSelectedCard={setSelectedCard}
+          message={messageInput.value}
+          onChangeMessage={messageInput.onChange}
+          errorMsg={messageInput.errorMsg}
+        />
+        <Divider spacing="0.5rem" fill={false} />
+        <Sender
+          senderInput={senderInput.value}
+          onChangeSenderInput={senderInput.onChange}
+          errorMsg={senderInput.errorMsg}
+        />
+        <Divider spacing="0.5rem" fill={false} />
+        <Recipient
+          name={recipientNameInput.value}
+          onChangeName={recipientNameInput.onChange}
+          errorMsgName={recipientNameInput.errorMsg}
+          phone={recipientPhoneInput.value}
+          onChangePhone={recipientPhoneInput.onChange}
+          errorMsgPhone={recipientPhoneInput.errorMsg}
+          quantity={recipientQuantityInput.value}
+          onChangeQuantity={recipientQuantityInput.onChange}
+          errorMsgQuantity={recipientQuantityInput.errorMsg}
+        />
+        <Divider spacing="0.5rem" fill={false} />
+        {product && <Product product={product} />}
+        <Divider spacing="3.125rem" />
+        <OrderBtn type="submit">{totalPrice}원 주문하기</OrderBtn>
+      </Content>
+    </Container>
+  );
+};
```

Comment:
```suggestion
const Order = () => {
  const orderContext = useOrderContext();
  return (
    <Container>
      <Content onSubmit={orderContext.submit}>
        <Order.Card />
        <Divider spacing="0.5rem" fill={false} />
        <Sender />
        <Divider spacing="0.5rem" fill={false} />
        <Order.Recipient />
        <Divider spacing="0.5rem" fill={false} />
        {product && <Product product={product} />}
        <Divider spacing="3.125rem" />
        <OrderBtn type="submit">{totalPrice}원 주문하기</OrderBtn>
      </Content>
    </Container>
  );
};
```

대략 이런 느낌이랄까..
Compound Component 라는 키워드로 검색해보면 나온답니다 ㅎㅎ